baklava/src/util.ts

```diff
@@ -0,0 +1,145 @@
+            PTT_STATUS = [false];
+        }
+        ioHook.on("keydown", event => {
+            if (event.shiftKey) {
+                if (CURRENT_PTT_KEY.includes("Shift")) {
+                    let i = CURRENT_PTT_KEY.indexOf("Shift");
+                    PTT_STATUS[i] = true;
+                }
+            } else if (event.altKey) {
+                if (CURRENT_PTT_KEY.includes("Alt")) {
+                    let i = CURRENT_PTT_KEY.indexOf("Alt");
+                    PTT_STATUS[i] = true;
+                }
+            } else if (event.ctrlKey) {
+                if (CURRENT_PTT_KEY.includes("Control")) {
+                    let i = CURRENT_PTT_KEY.indexOf("Control");
+                    PTT_STATUS[i] = true;
+                }
+            } else if (event.metaKey) {
+                if (CURRENT_PTT_KEY.includes("Meta")) {
+                    let i = CURRENT_PTT_KEY.indexOf("Meta");
+                    PTT_STATUS[i] = true;
+                }
+            } else {
+                if (CURRENT_PTT_KEY.includes(String.fromCharCode(event.rawcode))) {
+                    let i = CURRENT_PTT_KEY.indexOf(String.fromCharCode(event.rawcode));
+                    PTT_STATUS[i] = true;
+                }
+            }
+            let PTT = PTT_STATUS.every((key_status) => key_status === true);
+            if (PTT != PTT_PREV_STATUS) {
+                PTT_PREV_STATUS = PTT;
+                mainWindow.webContents.send("PTT_STATUS_CHANGE", !PTT);
+            }
+        })
+        ioHook.on("keyup", event => {
+            if (event.shiftKey) {
+                if (CURRENT_PTT_KEY.includes("Shift")) {
+                    let i = CURRENT_PTT_KEY.indexOf("Shift");
+                    PTT_STATUS[i] = false;
+                }
+            } else if (event.altKey) {
+                if (CURRENT_PTT_KEY.includes("Alt")) {
+                    let i = CURRENT_PTT_KEY.indexOf("Alt");
+                    PTT_STATUS[i] = false;
+                }
+            } else if (event.ctrlKey) {
+                if (CURRENT_PTT_KEY.includes("Control")) {
+                    let i = CURRENT_PTT_KEY.indexOf("Control");
+                    PTT_STATUS[i] = false;
+                }
+            } else if (event.metaKey) {
+                if (CURRENT_PTT_KEY.includes("Meta")) {
+                    let i = CURRENT_PTT_KEY.indexOf("Meta");
+                    PTT_STATUS[i] = false;
+                }
+            } else {
+                if (CURRENT_PTT_KEY.includes(String.fromCharCode(event.rawcode))) {
+                    let i = CURRENT_PTT_KEY.indexOf(String.fromCharCode(event.rawcode));
+                    PTT_STATUS[i] = false;
+                }
+            }
+            let PTT = PTT_STATUS.every((key_status) => key_status === true);
+            if (PTT != PTT_PREV_STATUS) {
+                PTT_PREV_STATUS = PTT;
+                mainWindow.webContents.send("PTT_STATUS_CHANGE", !PTT);
+            }
+        })
+        ioHook.start();
```

Comment:
do you need unsubscribe logic if someone binds push to talk multiple times?

Reply:
ahh thats a good point. mistake on my part will fix it

Reply:
lol wait i read your question wrong

Reply:
No you dont need unsub logic cuz im comparing the pressed key with a variable. but your question made me realize i was adding multiple `keyup` and `keydown` events *(whenever the user changed PTT keybind new listeners would be added)*. Im fixing this rn